### PR TITLE
xcm benchmarking for encointer runtime

### LINKED ISSUE #199: Encointer checks/improvements
## Check or fix

- [ ] setup `FeeManager` according to other SP parachains, is it ok to have it like `type FeeManager = ();` ?
- [x] `type IsReserve = NativeAsset;` - do other chains send `ReserveAssetDeposited` to Encointer?

## TODO

- [x] setup `pallet_xcm` benchmarks and replace `type WeightInfo = pallet_xcm::TestWeightInfo;`

cc: @brenzi

### PR DESCRIPTION
**Pull request by brenzi**
closes #199 

Add xcm benchmark setup for encointer-kusama-runtime

- [x] Does not require a CHANGELOG entry

### CODE REVIEW

**Comment by brenzi** [comment]
@bkontur With the example of the people chain, I added XCM benchmarks to encointer, but `pallet-xcm-benchmarks::generic` fails because of skipped benchmarks, whereas people accepts these skips. I seem to be missing some config.

File: system-parachains/encointer/src/lib.rs [hunk 1]
```diff
@@ -1120,7 +1278,7 @@ impl_runtime_apis! {
 			let params = (&config, &whitelist);
 			add_benchmarks!(params, batches);
 
-			if batches.is_empty() { return Err("Benchmark not found for this pallet.".into()) }
+			//if batches.is_empty() { return Err("Benchmark not found for this pallet.".into()) }
```
Comment (ggwpez) [inline system-parachains/encointer/src/lib.rs:1281]:
```suggestion
```

**Review by ggwpez — APPROVED** [review]

**Review by muharem — APPROVED** [review]

**Comment by bkchr** [comment]
@brenzi could you please format the code correctly? CI is not happy :D

**Comment by bkontur** [comment]
> @bkontur With the example of the people chain, I added XCM benchmarks to encointer, but `pallet-xcm-benchmarks::generic` fails because of skipped benchmarks, whereas people accepts these skips. I seem to be missing some config.
> 
> I did:
> 
> ```
> cargo build -p chain-spec-generator --profile production --features runtime-benchmarks
> ./target/production/chain-spec-generator --raw encointer-kusama-local > encointer-kusama-chain-spec.json
> ~/bin/polkadot-parachain-v1.14.0-runtime-benchmarks benchmark pallet --chain ./encointer-kusama-chain-spec.json --pallet=pallet-xcm-benchmarks::generic --extrinsic="*"
> ```
> 
> which throws errors. change `encointer-` to `people-` and it works
> 
> **edit: fixed in [7d2e67c](https://github.com/polkadot-fellows/runtimes/commit/7d2e67c80953111d692518f9561340cb1639b1ef)**

@brenzi can you merge this one please: https://github.com/encointer/runtimes/pull/1 ? It fixes the benchmarks for `pallet_xcm` which fail here: https://github.com/polkadot-fellows/runtimes/actions/runs/10528553477/job/29174892572?pr=440 
(and also I did fmt)

**Comment by brenzi** [comment]
@bkontur thanks for your fix. 
This also includes one-line changes in other systemchains. Was that on purpose? Seems out of context to this PR, but I don't mind

**Review by bkontur — APPROVED** [review]

**Pull request merged**